### PR TITLE
Add more properties to Capabilities class

### PR DESCRIPTION
**Pull request by huwdp**
TODO:

- [x] Add more info to ServerString.
- [x] Add other properties.
- [x] Test code
- [x] Clean-up code
- [x] Double check everything

This PR should eventually bring the total implemented amount over 80.310%.

### CODE REVIEW

File: src/scripting/flash/system/flashsystem.cpp [hunk 1]
```diff
@@ -176,6 +197,117 @@ ASFUNCTIONBODY_ATOM(Capabilities,_getServerString)
...
+{
+    LOG(LOG_NOT_IMPLEMENTED,"Capabilities.hasAudio always returns true");
+    ret = asAtomHandler::fromBool(true);
+}
+
+ASFUNCTIONBODY_ATOM(Capabilities,_hasAudioEncoder)
+{
+    LOG(LOG_NOT_IMPLEMENTED,"Capabilities.hasAudioEncoder always returns true");
+    ret = asAtomHandler::fromBool(true);
+}
```
Comment (dbluelle) [inline src/scripting/flash/system/flashsystem.cpp:211]: I think we should return false here as we don't have any support for AudioEncoding implemented.
The same goes for most of the following properties (except hasMP3 and hasTLS)

File: src/scripting/flash/system/flashsystem.cpp [hunk 2]
```diff
@@ -176,6 +197,117 @@ ASFUNCTIONBODY_ATOM(Capabilities,_getServerString)
 	*/
 	ret = asAtomHandler::fromString(sys,res);
 }
+
+ASFUNCTIONBODY_ATOM(Capabilities,_hasAudio)
+{
+    LOG(LOG_NOT_IMPLEMENTED,"Capabilities.hasAudio always returns true");
+    ret = asAtomHandler::fromBool(true);
+}
```
Comment (dbluelle) [inline src/scripting/flash/system/flashsystem.cpp:205]: We should return true only if we are compiling with ffmpeg enabled (#ifdef ENABLE_LIBAVCODEC)

File: src/scripting/flash/system/flashsystem.cpp [hunk 3]
```diff
@@ -176,6 +197,117 @@ ASFUNCTIONBODY_ATOM(Capabilities,_getServerString)
...
+ASFUNCTIONBODY_ATOM(Capabilities,_hasIME)
+{
+    LOG(LOG_NOT_IMPLEMENTED,"Capabilities.hasIME always returns true");
+    ret = asAtomHandler::fromBool(true);
+}
+
+ASFUNCTIONBODY_ATOM(Capabilities,_hasMP3)
+{
+    LOG(LOG_NOT_IMPLEMENTED,"Capabilities.hasMP3 always returns true");
+    ret = asAtomHandler::fromBool(true);
```
Comment (dbluelle) [inline src/scripting/flash/system/flashsystem.cpp:228]: We should return true only if we are compiling with ffmpeg enabled (#ifdef ENABLE_LIBAVCODEC)

File: src/scripting/flash/system/flashsystem.cpp [hunk 4]
```diff
@@ -139,6 +159,11 @@ ASFUNCTIONBODY_ATOM(Capabilities,_getServerString)
 		res += buf;
 	}
 
+    // Add hasAudio and hasMP3 to ServerString if available
+#ifdef ENABLE_LIBAVCODEC
+    res += "&A=t&MP3=t";
+#endif
```
Comment (dbluelle) [inline src/scripting/flash/system/flashsystem.cpp:165]: "A=t" and "MP3=t" are already in the default server string (see a few lines above) , I think we should remove them from there.
When that is done, we can merge this

**Comment by huwdp** [comment]
Thank you @dbluelle!

**Pull request merged**